src/main/io/serial_msp.c

```diff
@@ -400,25 +411,38 @@ static void serializeDataflashSummaryReply(void)
 }
 
 #ifdef USE_FLASHFS
-static void serializeDataflashReadReply(uint32_t address, uint8_t size)
+static void serializeDataflashReadReply(uint32_t address, uint16_t size, bool useLegacyFormat)
```

Comment:
this boolean argument makes the code much harder to follow.  perhaps two functions would be better here?